flatland/envs/rail_env.py

```diff
@@ -344,6 +350,9 @@ def reset(self, regenerate_rail: bool = True, regenerate_schedule: bool = True,
         self.agent_positions = np.zeros((self.height, self.width), dtype=int) - 1
         self._update_agent_positions_map(ignore_old_positions=False)
 
+        if self.effects_generator is not None:
+            self.effects_generator.post_reset(self)
```

Comment:
Should we pull up `EffectsGenerator` into `Environment` or keep in `RailEnv`? There might be more code to pull up?